xds/src/test/java/io/grpc/xds/internal/sds/SdsProtocolNegotiatorsTest.java

```diff
@@ -239,10 +245,11 @@ public SocketAddress localAddress() {
   }
 
   @Test
-  public void serverSdsHandler_nullTlsContext_expectPlaintext() throws IOException {
+  public void serverSdsHandler_nullTlsContext_expectFallbackProtocolNegotiator() {
     SdsProtocolNegotiators.HandlerPickerHandler handlerPickerHandler =
         new SdsProtocolNegotiators.HandlerPickerHandler(
-            grpcHandler, /* xdsClientWrapperForServerSds= */ null);
+            grpcHandler, /* xdsClientWrapperForServerSds= */ null,
+            new FallbackProtocolNegotiator());
```

Comment:
no reply for my previous comment. consider mock here. you can `verify` newHandler is called or not. this is cleaner and we can remove L316-344.

Reply:
Your previous comment was for a different file/test `XdsSdsClientServerTest#nullFallbackProtocolNegotiator_expectException` I renamed the test and also changed it. Also that test verifies the server closes so the clients can't connect any more.

For this one: yes, I will look into using a mock - I think it should work.